packages/ember-data/lib/system/model/model.js

```diff
@@ -781,6 +781,25 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
+  },
+
+  // handlers for the id property
+  unknownProperty: function (key) {
```

Comment:
I think this way requires uses to do `record.get('id')` to access the id. I was hoping to support `record.id` because it previously worked before the internal model refactor but maybe thats not possible :(.